connectors/sources/sharepoint.py

```diff
@@ -322,8 +322,12 @@ async def get_site_pages_content(
             return
 
         source_file_name = ""
+        format_filename = filename.replace("'", "''")
```

Comment:
this should be under `if self.is_cloud`
so the `'` char will be replaced by `%27%27` (just making sure that is the intent)

Reply:
Yes to ignore any single quote in mid of the url query .I am adding another single quote as a prefix. otherwise we are getting 400 Bad request error for the file name having single quote.